code/game/objects/effects/spawners/roguemapgen.dm

```diff
@@ -1,17 +1,46 @@
-/obj/effect/spawner/roguemap/Initialize(mapload)
-	..()
-	do_spawn()
-	return INITIALIZE_HINT_QDEL
-
+/obj/effect/spawner/roguemap
+	icon = 'icons/obj/structures_spawners.dmi'
+	///Chance to bother spawning anything
+	var/probby = 100
+	var/list/spawned		//a list of possible items to spawn e.g. list(/obj/item, /obj/structure, /obj/effect), can be weighted
+	var/lootmin = 1		//how many items will be spawned, at least.
+	var/lootmax = 1		//how many items will be spawned, at most
+	var/lootdoubles = TRUE	//if the same item can be spawned twice
+	var/fan_out_items = FALSE //Whether the items should be distributed to offsets 0,1,-1,2,-2,3,-3.. This overrides pixel_x/y on the spawner itself
 /obj/effect/spawner/roguemap/proc/do_spawn()
 	if(prob(probby))
-		var/obj/new_type = pick(spawned)
+		var/obj/new_type = pickweight(spawned)
 		new new_type(get_turf(src))
 
-/obj/effect/spawner/roguemap
-	icon = 'icons/obj/structures_spawners.dmi'
-	var/probby = 100
-	var/list/spawned
+/obj/effect/spawner/roguemap/Initialize(mapload)
+	..()
+	if(!prob(probby))
+		return INITIALIZE_HINT_QDEL
+	if(spawned && spawned.len > 1)
+		var/turf/T = get_turf(src)
+		var/loot_spawned = 0
+		var/chosenamout = rand(lootmin,lootmax)
+		while((chosenamout-loot_spawned) && spawned.len)
```

Comment:
this has no sanity check, why is it a while loop over a for loop?

Reply:
I will not lie, I copied and edited this from another obj/effect/spawner and tested if it works and it did.

I'll just fix them both real quick...